src/main/java/de/hhu/stups/plues/ui/controller/BatchTimetableGeneration.java

```diff
@@ -0,0 +1,338 @@
+import java.util.List;
+import java.util.ResourceBundle;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+
+public class BatchTimetableGeneration extends GridPane implements Initializable {
+
+  private final Logger logger = Logger.getLogger(getClass().getSimpleName());
+  private final Delayed<Store> delayedStore;
+  private final Delayed<SolverService> delayedSolverService;
+
+  private final BooleanProperty solverProperty;
+  private final BooleanProperty generationStarted;
+  private final BooleanProperty generationSucceeded;
+  private final BatchResultBoxFactory resultBoxFactory;
+  private Path tempDirectoryPath;
+  private Set<PdfRenderingTask> taskPool;
+  private Set<PdfRenderingTask> executableTaskPool;
+  private Set<BatchResultBox> boxPool;
+  private final ExecutorService executor;
+  private Task<Void> fillPoolTask;
+  private Task<Void> executePoolTask;
+
+  @FXML
+  @SuppressWarnings("unused")
+  private Button btGenerateAll;
+  @FXML
+  @SuppressWarnings("unused")
+  private Button btSaveToZip;
+  @FXML
+  @SuppressWarnings("unused")
+  private Button btSaveToFolder;
+  @FXML
+  @SuppressWarnings("unused")
+  private Button btCancel;
+  @FXML
+  @SuppressWarnings("unused")
+  private ListView<BatchResultBox> listView;
+
+  /**
+   * Generate all possible combinations of major and minor courses. While generating the pdf files
+   * save them in a temporary directory. When all tasks are finished the user is able to store the
+   * pdf files persistently in a folder or a zip archive.
+   */
+  @Inject
+  public BatchTimetableGeneration(final FXMLLoader loader, final Delayed<Store> delayedStore,
+                                  final Delayed<SolverService> delayedSolverService,
+                                  final BatchResultBoxFactory resultBoxFactory,
+                                  final ExecutorService executorService) {
+    this.delayedStore = delayedStore;
+    this.delayedSolverService = delayedSolverService;
+    this.resultBoxFactory = resultBoxFactory;
+    this.executor = executorService;
+
+    this.solverProperty = new SimpleBooleanProperty(false);
+    this.generationStarted = new SimpleBooleanProperty(false);
+    this.generationSucceeded = new SimpleBooleanProperty(false);
+    this.taskPool = new HashSet<>();
+    this.executableTaskPool = new HashSet<>();
+    this.boxPool = new HashSet<>();
+
+    try {
+      this.tempDirectoryPath = Files.createTempDirectory("plues_timetables");
+    } catch (final IOException exception) {
+      logger.log(Level.INFO, "Could not create temporary directory.");
+    }
+
+    // delete the temporary directory when the application closes
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```

Comment:
As it is a tmp directory I think we can just ignore it after the application closes